### ISSUE TITLE
Formalise extended list of URLs

### ISSUE THREAD

**Issue by judgej** [thephpleague/omnipay#280]
At the moment a `cancelUrl` and `returnUrl` is defined when creating a request message. The `returnUrl` is used by some gateways as a notification URL, but other gateways define a different parameter for this. Extending the list of URLs that OmniPay will accept, and being stricter on their use, would help with consistency.

Here are those I know of:

* `returnUrl` - the URL the gateway should redirect the user to. Some gateways define this in the account details, some when creating an authorisation request, and some are supplied by the notification handler.
* `notifyUrl` - the URL where the gateway should send its results to, when the results are handled by a notification handler (back-channel, out of the user's session). Some gateways require this to be pre-defined in the account, and some can be generated dynamically (so can take GET parameters where that helps).
* `cancelUrl` - where the gateway should redirect the user to if the user decides to cancel the transaction before it is authorised. Some gateways define this in the gateway account, and some allow it to be defined in code.

Any other URLs that need to be passed to gateway drivers?

**Comment by aimeos** [comment]
+1

**Comment by kayladnls** [comment]
+1

**Comment by yellow1912** [comment]
+1

**Comment by makasim** [comment]
+1

**Comment by barryvdh** [comment]
So this would be a matter of just adding it to the docs and possibly renaming some gateways when upgrading, right? No actual code changes needed?

**Comment by judgej** [comment]
Yes, I think so. It's a case of saying, "If your gateway supports a URL of type X, then *this* is what you call it".
